src/javaharness/java/arcs/api/ParticleBase.java

```diff
@@ -60,57 +62,19 @@ public void onHandleDesync(Handle handle) {
   }
 
   @Override
-  public SlotProxy getSlot(String name) {
-    return slotProxyByName.get(name);
+  public void setOutput(Consumer<PortableJson> output) {
+    this.output = output;
   }
 
   @Override
-  public boolean hasSlotProxy(String name) {
-    return slotProxyByName.containsKey(name);
-  }
-
-  @Override
-  public void addSlotProxy(SlotProxy slotProxy) {
-    slotProxyByName.put(slotProxy.slotName, slotProxy);
-  }
-
-  @Override
-  public void removeSlotProxy(String name) {
-    slotProxyByName.remove(name);
-  }
-
-  @Override
-  public void renderSlot(String slotName, List<String> contentTypes) {
-    SlotProxy slot = getSlot(slotName);
-    if (slot == null) {
-      // Did not receive StartRender
-      return;
+  public void renderModel() {
+    if (this.output != null) {
+      this.output.accept(jsonParser.emptyObject()
+          .put("template", getTemplate(""))
+          .put("model", getModel()));
     }
-    slot.requestedContentTypes.addAll(contentTypes);
-    StringBuilder content = new StringBuilder("{");
-    if (shouldRender(slotName)) {
-      content.append("\"templateName\":\"").append(getTemplateName(slotName)).append("\", ");
-      if (slot.requestedContentTypes.contains("template")) {
-        content.append("\"template\":\"").append(getTemplate(slotName)).append("\", ");
-      }
-      if (slot.requestedContentTypes.contains("model")) {
-        content.append("\"model\":").append(getModel());
-      }
-    }
-    content.append("}");
-    slot.render(content.toString());
-  }
-
-  @Override
-  public boolean shouldRender(String slotName) {
-    return true;
   }
-
-  @Override
-  public String getTemplateName(String slotName) {
-    return "default";
-  }
-
+ 
   @Override
   public String getTemplate(String slotName) {
     return "";
```

Comment:
What's the meaning of "" for slot name?

Reply:
In theory particle may render into several slots:
```
particle MyParticle
  consume slot1
  consume slot2
```
so getTemplate("slot1") or getTemplate("slot2") would return a different template for the appropriate slot.
In reality of the new rendering implementation in js, multiple slots per particle are not supported yet. I replicated that implementation, so I'm just passing an empty string as slot name. Added a todo to add support for this later.